testing-guidance/workflow-tests/scripting/lua-scripting-tests.md

```diff
@@ -0,0 +1,293 @@
+| Create Lua scripts utilizing the Script Event to send a variable from one to another | <ol><li>Create two new Lua scripts and copy the contents of *SendEvent.lua* and *ReceiveEvent.lua* found under the **"Area: Using ScriptEvents to communicate between two Lua scripts" scripts:** to them</li><li>Save the Lua scripts with their respective names: *SendEvent.lua* and *ReceiveEvent.lua*</li></ol> | <ul><li>New Lua scripts can be created</li><li>Lua scripts save successfully with provided code</li></ul> |
+| Create entities using the Lua scripts and enter Game Mode | <ol><li>Create two entities with Lua Script component added to them</li><li>Assign *SendEvent.lua* to one of the entities and *ReceiveEvent.lua* to the other</li><li>Enter Game Mode</li></ol> | <ul><li>Entities with Lua Script components can be created</li><li>Lua scripts can be assigned to the Lua Script component</li><li>Assigned scripts execute printing *Received number: 123.0* about every second</li></ul> |
+
+**"Area: Using ScriptEvents to communicate between two Lua scripts" scripts:**
+
+*SendEvent.lua*
+```
+local SendEvent = 
+{
+}
+
+function SendEvent:OnActivate()
+	--Connect to Tick bus, so that OnTick(dt) function is available further on
+	self.tickHandler = TickBus.Connect(self)
+	--Create a timer variable with value of 1, later used in the delay setup
+	timer = 1
+	--Connect to the created Script Event (SimpleScriptEvent is the name of the Script Event file)
+	self.ScriptEventHandler = SimpleScriptEvent.Connect(self)
+end
+
+--The delay setup is put in place so that the Send event is guaranteed to launch after all other entities/scripts (including the Receive event) are active
+
+--OnTick is activated each tick returning delta time (dt) since the last tick
+function SendEvent:OnTick(dt)
+	--Each tick subtract the time it took since the last tick from the current timer value and overwrite it (Can be looked at as "Count from 1 to 0 using ticks")
+	timer = timer - dt
+	--If timer variable reaches 0 or negative values then
+	if timer <= 0 then
+		--Send the SimpleScriptEvent's SendNumber event with a Number parameter 123
+		SimpleScriptEvent.Broadcast.SendNumber(123)
+		--Reset the timer variable to 1
+		timer = 1
+	end
+end
+
+return SendEvent
+```
+*ReceiveEvent.lua*
+```
+local ReceiveEvent = 
+{
+}
+
+function ReceiveEvent:OnActivate()
+	--Connect to the created Script Event (SimpleScriptEvent is the name of the Script Event file)
+	self.ScriptEventHandler = SimpleScriptEvent.Connect(self, self.entityId)
+
+end
+
+--Receive the SimpleScriptEvent's SendNumber event with it's parameter (number)
+function ReceiveEvent:SendNumber(number)
+	--Print the received parameter to the Console
+	Debug.Log("Received number: " .. tostring(number))
+end
+
+return ReceiveEvent
+```
```

Comment:
Let's put the code here in a "collapsed section".

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

Reply:
Addressed in earlier point.